hyperspy/_components/power_law.py

```diff
@@ -109,12 +109,14 @@ def estimate_parameters(self, signal, x1, x2, only_current=False,
             s = signal.get_current_signal()
         else:
             s = signal
-        I1 = s.isig[i1:i3].integrate1D(2j).data.astype("float")
-        I2 = s.isig[i3:i2].integrate1D(2j).data.astype("float")
         if s._lazy:
             import dask.array as da
             log = da.log
+            I1 = s.isig[i1:i3].integrate1D(2j).data.astype("float")
```

Comment:
I would rather keep this code the same for ``lazy`` and standard signals since they should both give the same result.

Although proper integration vs simply summing doesn't make much difference in your example, it can be very different in other cases for unbinned signals e.g. for non-linear axes. I think that it'll be better to leave it as it was.

Reply:
Shall we just do the simple sum for binned signal?

Reply:
Yes, but it does it ~50-100 times slower! I have changed to use the `out` parameter but it is still not as fast as the faster alternative (possibly a factor 10 slower). I may look at further improvement but for now, I guess this is good enough.

Reply:
That's interesting and it reminds me of the important overhead involved in deepcopying a signal because of the ``metadata``. Something to take seriously elsewhere.